tests/semantic/cmp/cmp.dl

```diff
@@ -0,0 +1,21 @@
+// Souffle - A Datalog Compiler
+// Copyright (c) 2020, The Souffle Developers. All rights reserved
+// Licensed under the Universal Permissive License v 1.0 as shown at:
+// - https://opensource.org/licenses/UPL
+// - <souffle root>/licenses/SOUFFLE-UPL.txt
+
+// Index comparator test for attributes of different type
```

Comment:
The comment could be expanded - perhaps explaining that the test is for indirect index generation which is expected to change at 6+ attributes.

 Currently this tests indirect indices, but a change in the code would leave this testing direct indices and there would be no indication that this test was no longer fulfilling its purpose.

Reply:
I renamed the files and added the comment.
Unfortunately, constructing a test case with more attributes is quite involved. 
We would need to construct an example for which the RamDomain (int32_t) comparator states exactly the opposite of the attribute comparator.